[stable/ghost] Replace initContainer by securityContext

tompizmor opened the pull request:
Signed-off-by: tompizmor <tompizmor@gmail.com>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
The ghost chart was using an initContainer to change the permissions of the persistent volume as the ghost container is a non-root one. This worked fine on standard Kubernetes distributions but it does not work on Openshift as all the containers should run as non-root. 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

tompizmor committed:
[stable/ghost] Replace initContainer by securityContext

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/ghost] Replace initContainer by securityContext

Signed-off-by: tompizmor <tompizmor@gmail.com>

davidkarlsen committed:
upgrade auditbeat (#8277)

Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>

alexandrst88 committed:
Replaced toml block for traefik logs (#8288)

Signed-off-by: Oleksandr Stepanov <alexandrst88@gmail.com>

davidkarlsen committed:
[stable/external-dns] upgrade external-dns to 0.5.7 (#8275)

Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>

javsalgar committed:
[stable/opencart] Update NOTES.txt setting the database password (#8291)

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

javsalgar committed:
[stable/magento] Update NOTES.txt setting the database password on upgrade (#8290)

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

javsalgar committed:
[stable/osclass] Update NOTES.txt setting the database password (#8292)

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

oliverisaac committed:
Spinnaker inline script in values (#8234)

* Added option for hal config to be created by the values.yaml file

Signed-off-by: Oliver Isaac <oisaac@gmail.com>

* Added support for creating scripts, config maps, and secrets from the values.yaml file. Included an example in values_saml.yaml on how to use this functinality to implement saml authentication.

Signed-off-by: Oliver Isaac <oisaac@gmail.com>

* Version bump

Signed-off-by: Oliver Isaac <oisaac@gmail.com>

Commit:
[stable/phabricator] Release 3.0.6 (#8296)

* phabricator: update to `2018.40.0-debian-9`

Signed-off-by: Bitnami Containers <containers@bitnami.com>

* phabricator: updated chart requirements

Signed-off-by: Bitnami Containers <containers@bitnami.com>

* phabricator: bump chart appVersion to `2018.40.0`

Signed-off-by: Bitnami Containers <containers@bitnami.com>

* phabricator: bump chart version to `3.0.6`

Signed-off-by: Bitnami Containers <containers@bitnami.com>

sstarcher committed:
update newrelic infra to 1.2.0 (#8301)

Signed-off-by: Shane Starcher <shanestarcher@gmail.com>

desaintmartin committed:
[incubator/elasticsearch] Add nodeAffinity parameters (#7567)

* [incubator/elasticsearch] Fix missing *.antiAffinity definitions in README.

Signed-off-by: Cédric de Saint Martin <cdesaintmartin@wiremind.fr>

* [incubator/elasticsearch] Add nodeAffinity parameters, empty by default.

Signed-off-by: Cédric de Saint Martin <cdesaintmartin@wiremind.fr>

mschuchard committed:
Add Kibana access notes to output (#8132)

Signed-off-by: Matt Schuchard <grapheneowns@gmail.com>

anthonydahanne committed:
Initial commit for Terracotta Server OSS helm chart (#8273)

* Initial commit for Terracotta Server OSS helm chart

Signed-off-by: Anthony Dahanne <anthony.dahanne@gmail.com>

* Add new line character at end of file, to pass validation

Signed-off-by: Anthony Dahanne <anthony.dahanne@gmail.com>

* Better formatting, as suggested during PR review

Signed-off-by: Anthony Dahanne <anthony.dahanne@gmail.com>

* Add RBAC

Signed-off-by: Anthony Dahanne <anthony.dahanne@gmail.com>

faizanahmad055 committed:
[Stable] Add support for configmap, additional volumes and volume mounts sonatype-nexus chart (#8226)

* Add support for configmap and additional volumes in sonatype-nexus

Signed-off-by: faizanahmad055 <faizan.ahmad55@outlook.com>

* Add new line at the end of values.yaml file

Signed-off-by: faizanahmad055 <faizan.ahmad55@outlook.com>

* Remove additional volumes and volumeMounts and map configmap if enabled

Signed-off-by: faizanahmad055 <faizan.ahmad55@outlook.com>

* Bump the minor version

Signed-off-by: faizanahmad055 <faizan.ahmad55@outlook.com>

* Update readme

Signed-off-by: faizanahmad055 <faizan.ahmad55@outlook.com>

guptaarvindk committed:
Added support for clusterIP and headless services. (#8312)

Signed-off-by: Arvind Gupta <guptaarvindk@gmail.com>

cpanato committed:
add Jaeger-operator to helm charts (#8107)

Signed-off-by: cpanato <ctadeu@gmail.com>

sebastien-prudhomme committed:
[incubator/elasticsearch] fix hardcoded busybox image name (#8082)

* [incubator/elasticsearch] fix hardcoded busybox image name

Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

* [incubator/elasticsearch] fix readme

Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

* [incubator/elasticsearch] fix Chart version

Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

* [incubator/elasticsearch] best practice for Busybox image

Also fix the default pull policy for the main container in the readme

Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

* [incubator/elasticsearch] chart version bump

Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

* [incubator/elasticsearch] better parameter names

Signed-off-by: Sébastien Prud'homme <sebastien.prudhomme@gmail.com>

juan131 reviewed (changes requested):
We should keep the **initContainer** since there're some PV provisioners do not support **securityContext**

tompizmor committed:
[stable/mongodb] - Allow using custom initialization scripts (#8272)

* [stable/mongodb] - Allow using custom initialization scripts

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Avoid creating config-map if not needed

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Delete blank line

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Add missing conditional blocks

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
[stable/mariadb] Expose securityContext parameters in values yaml files (#8287)

* [stable/mariadb] Expose securityContext parameters in values yaml files

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Fix capitalization

Signed-off-by: tompizmor <tompizmor@gmail.com>

* Fix capitalization - Part II

Signed-off-by: tompizmor <tompizmor@gmail.com>

andrenarchy committed:
[stable/stellar-core]: Release 1.0.0 (initial release) (#7266)

* [incubator/stellar-core]: Release 0.1.0 (initial release)

Signed-off-by: André Gaul <andre@gaul.io>

* [incubator/stellar-core]: use http readiness probe

Signed-off-by: André Gaul <andre@gaul.io>

* [incubator/stellar-core]: update NOTES

Signed-off-by: André Gaul <andre@gaul.io>

* [incubator/stellar-core]: update requirements.yaml with repo url

Signed-off-by: André Gaul <andre@gaul.io>

* [incubator/stellar-core]: add rendhalver as maintainer/owner

Signed-off-by: André Gaul <andre@gaul.io>

* [incubator/stellar-core]: properly handle postgresql name override in helpers

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: move stellar-core to stable

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: adapt install command in README

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: enable 10.0.0-2 features (flavors, archive init)

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: use deployment strategy recreate

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: add serviceaccount

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: add default timeout for postgres

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: fix versions

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: update postgresql

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: fix version in readme

Signed-off-by: André Gaul <andre@gaul.io>

* [stable/stellar-core]: fix bug in secret template

Signed-off-by: André Gaul <andre@gaul.io>

javsalgar committed:
[stable/mariadb] Enable metrics by default in production environment (#8322)

* [stable/mariadb] Enable metrics by default in production environment

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

* Bump chart version

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

yaroslavya committed:
typo fixed (#8068)

are going are going-> are going

Signed-off-by: Yaroslav Yakovlev <yaroslavya@gmail.com>

updated chart version

davidkarlsen committed:
Upgrade grafana (#8328)

* upgrade to grafana 5.2.5

Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>

* upgrade

Signed-off-by: David J. M. Karlsen <david@davidkarlsen.com>

Commit:
[stable/joomla] Add nodeSelector, tolerations and affinity (#8319)

Signed-off-by: juan131 <juan@bitnami.com>

Commit:
[stable/wordpress] Simplify ingress configuration for cert-manager (#8335)

* [stable/wordpress] Simplify ingress configuration when for cert-manager

Signed-off-by: juan131 <juan@bitnami.com>

* Apply changes on 'values-production.yaml' too

Signed-off-by: juan131 <juan@bitnami.com>

Commit:
[stable/mongodb-replicaset] Fix various bootstrap issues (#8115)

Signed-off-by: Steven Sheehy <ssheehy@firescope.com>

unguiculus committed:
[stable/keycloak] Update to 4.5.0.Final (#8192)

* The Docker image has added support for DNS_PING which is now used
  instead of JDBC_PING
* The StatefulSet is updated to `apps/v1`

Signed-off-by: Reinhard Nägele <unguiculus@gmail.com>

bhavin192 committed:
[stable/prometheus-adapter] Add checksum of configMap as annotation (#8334)

* [stable/prometheus-adapter] Add checksum of configMap as annotation

- This adds checksum/config to the annotations which has sha256sum of
  the file custom-metrics-configmap.yaml
- This does the rolling update of the deployment when user does helm
  upgrade and there are changes to the configMap, so the adapter will
  pick up the latest changes in the config file
- ref:
  https://docs.helm.sh/charts_tips_and_tricks/#automatically-roll-deployments-when-configmaps-or-secrets-change
- Bump chart version

Signed-off-by: Bhavin Gandhi <bhavin7392@gmail.com>

* [stable/prometheus-adapter] Bump version to 0.2.0

Signed-off-by: Bhavin Gandhi <bhavin7392@gmail.com>

mschuchard committed:
Fix elastic DNS information output. (#8131)

Signed-off-by: Matt Schuchard <grapheneowns@gmail.com>

javsalgar committed:
[stable/rabbitmq] Log to stdout and add metrics exporter (#8337)

* [stable/rabbitmq] Log to stdout and add metrics exporter

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

* Update readme

Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

Commit:
[stable/joomla] Release 3.1.1 (#8354)

* joomla: update to `3.8.13-debian-9`

Signed-off-by: Bitnami Containers <containers@bitnami.com>

* joomla: updated chart requirements

Signed-off-by: Bitnami Containers <containers@bitnami.com>

* joomla: bump chart appVersion to `3.8.13`

Signed-off-by: Bitnami Containers <containers@bitnami.com>

* joomla: bump chart version to `3.1.1`

Signed-off-by: Bitnami Containers <containers@bitnami.com>

Commit:
[stable/percona] Add support for nodeSelector and tolerations labels (#7067)

This commit adds support for the following scheduling labels:

- nodeSelector:
  https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
- tolerations:
  https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/

These couple of labels will allow custom pods scheduling depending on
specified labels. By default, both labels will be empty.

Signed-off-by: Obed N Munoz <obed.n.munoz@intel.com>

tompizmor committed:
Use initContainer if securityContext is disables

Signed-off-by: tompizmor <tompizmor@gmail.com>

tompizmor committed:
Fix conflicts

Signed-off-by: tompizmor <tompizmor@gmail.com>
